eox_nelp/pearson_vue/pipeline.py

```diff
@@ -496,3 +498,104 @@ def raise_audit_pearson_exception(exception_dict, failed_step_pipeline):
+              enrollment ID, anonymous user model ID, and anonymous user ID.
+    """
+    client_authorization_id = request_data["authorization"]["clientAuthorizationID"]
+    client_authorization_elements = client_authorization_id.split("-")
+    enrollment_id = client_authorization_elements[0]
+    anonymous_user_model_id = client_authorization_elements[1]
+    anonymous_user_id = request_data["clientCandidateID"].replace("NELC", "")
+
+    return {
+        "exam_result": request_data["exams"]["exam"][0]["examResult"],
+        "client_authorization_id": client_authorization_id,
+        "enrollment_id": enrollment_id,
+        "anonymous_user_model_id": anonymous_user_model_id,
+        "anonymous_user_id": anonymous_user_id,
+    }
+
+
+def generate_external_certificate(enrollment, exam_result, **kwargs):  # pylint: disable=unused-argument
+    """
+    Generates an external certificate if the exam result meets the passing criteria.
+
+    Args:
+        enrollment (object): The enrollment object associated with the user and course.
+        exam_result (dict): The dictionary containing the exam result data.
+        **kwargs: Additional keyword arguments.
+    """
+    passing_score = float(exam_result.get("passingScore", 1))
+    score = float(exam_result.get("score", 0))
+
+    if score >= passing_score:
+        generate_course_certificate(
+            enrollment.user,
+            enrollment.course_id,
+            "downloadable",
+            enrollment.mode,
+            score,
+            "batch"
+        )
+
+
+def get_enrollment_from_anonymous_user_id(
+    anonymous_user_model_id,
+    enrollment=None,
+    **kwargs
+):  # pylint: disable=unused-argument
+    """
+    Retrieves enrollment information based on the anonymous user model ID.
+
+    Args:
+        anonymous_user_model_id (str): The ID of the anonymous user model.
+        enrollment (object, optional): An optional enrollment object.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        dict: A dictionary containing the enrollment object if found, otherwise an empty dictionary.
+    """
+    if not enrollment:
+        try:
+            anonymous_user_id = AnonymousUserId.objects.get(id=anonymous_user_model_id)
+
+            return {
+                "enrollment": CourseEnrollment.objects.get(
+                    user=anonymous_user_id.user,
+                    course=anonymous_user_id.course_id,
+                )
+            }
+        except ObjectDoesNotExist:
+            pass
+
+    return {}
+
+
+def get_enrollment_from_id(enrollment_id, **kwargs):  # pylint: disable=unused-argument
+    """
+    Retrieves enrollment information based on the enrollment ID.
+
+    Args:
+        enrollment_id (str): The ID of the enrollment.
+        **kwargs: Additional keyword arguments.
+
+    Returns:
+        dict: A dictionary containing the enrollment object if found, otherwise an empty dictionary.
+    """
```

Comment:
Could you add the block code:
``` python3
if not enrollment:
```
In order to not be concerned it is first 
`get_enrollment_from_anonymous_user_id` or `get_enrollment_from_id`